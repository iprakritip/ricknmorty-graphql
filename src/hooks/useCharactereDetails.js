import { gql, useQuery } from '@apollo/client'
import React from 'react'

export const GET_CHARACTER_DETAILS = gql`
query GetCharacterDetails($id:ID!){
    character(id: $id) {
    id
    name
    image
    status
    gender
    species
    created
    episode {
        id
      name
      episode
    }
    origin{
      id
      name
    }
    location{
      id
      name
    }
  }
}
`
export const useCharacterDetails = (id) => {
  const { data, loading, error } = useQuery(GET_CHARACTER_DETAILS, {
    variables: {
      id
    }
  });
  return { data, loading, error }
}
