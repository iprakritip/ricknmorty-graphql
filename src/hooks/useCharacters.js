import { gql, useQuery } from '@apollo/client'
import React from 'react'


const GET_CHARACTERS = gql`
query getCharacters{
    characters{
        results{
            id
            name
            image
            species
        }
    }
}
`
export const useCharacters=()=> {
    const { error, loading, data } = useQuery(GET_CHARACTERS);
    return { error, loading, data }
}


