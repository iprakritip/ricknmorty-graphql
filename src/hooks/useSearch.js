import { gql, useLazyQuery } from '@apollo/client'
import React from 'react'

const GET_CHARACTER_LOCATIONS = gql`
query GetLocations($name: String!){
    characters(filter: 
    {name: $name
    }) {
    results {
      location {
        name
      }
    }
  }
}
`

export const useSearch = (name) => {
    const [getLocations, { data, loading, error }] = useLazyQuery(GET_CHARACTER_LOCATIONS, {
        variables: {
            name
        }
    })
    return { getLocations, data, loading, error }
}
