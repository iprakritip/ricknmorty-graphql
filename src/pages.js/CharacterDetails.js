import React from 'react'
import { useParams } from 'react-router-dom'
import CharacterProfile from '../components/CharacterProfile';
import Error from '../components/Error';
import Loading from '../components/Loading';
import { useCharacterDetails } from '../hooks/useCharactereDetails'

export default function CharacterDetails() {
    const { id } = useParams();
    const { data, loading, error } = useCharacterDetails(id)
    console.log(data);

    if (loading) return <Loading />
    if (error) return <Error />
    if (!data.character) return <Error />
    return (
        <CharacterProfile character={data.character} />
    )
}
