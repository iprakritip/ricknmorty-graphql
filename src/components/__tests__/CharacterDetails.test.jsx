import { MockedProvider } from "@apollo/client/testing";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import { GET_CHARACTER_DETAILS } from "../../hooks/useCharactereDetails";
import CharacterDetails from "../../pages.js/CharacterDetails";


jest.mock("react-router-dom", () => ({
    ...(jest.requireActual("react-router-dom")),
    useParams: () => ({ id: 7 })
}));

const mocks = [{
    request: {
        query: GET_CHARACTER_DETAILS,
        variables: {
            id: 7
        }
    },
    
  "result": {
    data:{
    "character": {
      "id": "7",
      "name": "Abradolf Lincler",
      "image": "https://rickandmortyapi.com/api/character/avatar/7.jpeg",
      "status": "unknown",
      "gender": "Male",
      "species": "Human",
      "created": "2017-11-04T19:59:20.523Z",
      "episode": [
        {
          "id": "10",
          "episode": "S01E10"
        },
        {
          "id": "11",
          "episode": "S01E11"
        }
      ],
      "origin": {
        "id": "20",
        "name": "Earth (Replacement Dimension)"
      },
      "location": {
        "id": "21",
        "name": "Testicle Monster Dimension"
      }
      }
  }}
}]
const TestComponent=(<MockedProvider mocks={mocks} >
    <CharacterDetails />
</MockedProvider>)

test('should render a profile', async () => {
    render(TestComponent)
    expect(await screen.findByRole('heading', {level:2})).toBeInTheDocument();
})

it('matches snapshot',async () => {
    const {baseElement} = render(TestComponent);
    expect(await screen.findByRole('heading', {level:2})).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });




