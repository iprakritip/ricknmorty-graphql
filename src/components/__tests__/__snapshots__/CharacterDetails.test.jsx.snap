// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<body>
  <div>
    <div
      class="characterProfile"
    >
      <img
        alt=""
        height="500"
        src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
        width="500"
      />
      <div
        class="characterDetails"
      >
        <h2>
          Abradolf Lincler
        </h2>
        <div>
          <p>
            <b>
              S01E10
            </b>
             - 
            Close Rick-counters of the Rick Kind
          </p>
        </div>
        <div>
          <p>
            <b>
              S01E11
            </b>
             - 
            Ricksy Business
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
